Sources/NIOCore/CircularBuffer.swift

```diff
@@ -222,6 +222,33 @@ extension CircularBuffer: Collection, MutableCollection {
             return end.backingIndex &- start.backingIndex
         }
     }
+
+    @inlinable
+    public __consuming func _copyContents(
+        initializing buffer: UnsafeMutableBufferPointer<Element>
+    ) -> (Iterator,UnsafeMutableBufferPointer<Element>.Index) {
+        precondition(buffer.count >= self.count)
+
+        guard var ptr = buffer.baseAddress else {
+            return (self.makeIterator(), buffer.startIndex)
+        }
+
+        let indexRanges: [Range<Int>]
```

Comment:
Hmm, we need to confirm that this doesn't allocate, but I forgot to get you to set limits. One sec.